transcom/mymove
# INT B-19806 - Back End - SC ability to edit ‘Expense Type’ & "Description" fields

## [B-19806](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-19806)

## Summary

This branch adds the backend functionality needed for an office member to create and delete weight tickets, progear weight tickets, and moving expenses. The ability to edit/PATCH these objects were already available and did not need to be added.

The frontend branch for this ticket will use these for the "Expense Type" portion of the story. When clicking the button to convert an expense type from one to another, it will use the new delete endpoint to soft delete the existing expense in the DB, and the new create endpoint to create a new expense of the correct type. Since description is just a PATCH request, no work was needed for that on this branch.

### How to test

1. If using Postman, after building this branch, import the new GHC .yaml file from the `swagger` directory. the new endpoints are all under `ppm-shipments/{ppmShipmentId}`. They are `weight-ticket`, `moving-expenses`, and `pro-gear-weight-tickets`. You should see a POST at the top level for each of these, and a DELETE when you drill down to the submenu where you need the expense's ID.
2. Login as a SC, go to a move that has at least on of each expense type (I used the move with code PPMRFC in local dev)
3. In the network tab, get the ppm shipment's id from the `mto_shipments` network response.
4. Also, you will need to add an `X-CSRF-Token` Header for each Postman request, this will just be the "masked_gorilla" cookie value.
5. Issue a DELETE request for each expense type. Now, when you check for that expense in it's table (`moving_expenses`, `weight_tickets`, `progear_weight_tickets`), scroll to the right until you see the `deleted_at` column. This should be populated with a timestamp corresponding to when you sent the request.
![Screenshot 2024-05-09 at 1 01 19 PM](https://github.com/transcom/mymove/assets/146959895/52448c0d-84b5-40bf-8064-a8aadc6f908f)

![Screenshot 2024-05-09 at 1 01 51 PM](https://github.com/transcom/mymove/assets/146959895/a4a1a731-3cfb-4a9c-b936-db36023f4f6c)

6. Now issue a POST request for each document type, you should get a response that includes the new document's ID. Go into DBeaver and verify that the new document has been added (remember to refresh DBeaver with the button at the bottom left).
